blivet/fstab.py

```diff
@@ -0,0 +1,279 @@
+                                              exists=True,
+                                              device=spec))
+        elif spec.startswith("/") and vfstype == "swap":
+            # swap file
+            device = FileDevice(spec,
+                                parents=self._get_containing_device(spec, blivet.devicetree),
+                                fmt=get_format(vfstype,
+                                               device=spec,
+                                               exists=True),
+                                exists=True)
+        elif vfstype == "bind" or "bind" in mntops:
+            # bind mount... set vfstype so later comparison won't
+            # turn up false positives
+            vfstype = "bind"
+
+            # This is probably not going to do anything useful, so we'll
+            # make sure to try again from FSSet.mount_filesystems. The bind
+            # mount targets should be accessible by the time we try to do
+            # the bind mount from there.
+            parents = self._get_containing_device(spec, blivet.devicetree)
+            device = DirectoryDevice(spec, parents=parents, exists=True)
+            device.format = get_format("bind",
+                                       device=device.path,
+                                       exists=True)
+        elif file in ("/proc", "/sys", "/dev/shm", "/dev/pts",
+                      "/sys/fs/selinux", "/proc/bus/usb", "/sys/firmware/efi/efivars"):
+            # drop these now -- we'll recreate later
+            return None
+        else:
+            # nodev filesystem -- preserve or drop completely?
+            fmt = get_format(vfstype)
+            fmt_class = get_device_format_class("nodev")
+            if spec == "none" or \
+               (fmt_class and isinstance(fmt, fmt_class)):
+                device = NoDevice(fmt=fmt)
+
+        if device is None:
+            log.error("failed to resolve %s (%s) from fstab", spec,
+                      vfstype)
+            raise ValueError()
+
+        device.setup()
+        fmt = get_format(vfstype, device=device.path, exists=True)
+        if vfstype != "auto" and None in (device.format.type, fmt.type):
+            log.info("Unrecognized filesystem type for %s (%s)",
+                     device.name, vfstype)
+            device.teardown()
+            raise ValueError()
+
+        # make sure, if we're using a device from the tree, that
+        # the device's format we found matches what's in the fstab
+        ftype = getattr(fmt, "mount_type", fmt.type)
+        dtype = getattr(device.format, "mount_type", device.format.type)
+        if hasattr(fmt, "test_mount") and vfstype != "auto" and ftype != dtype:
+            log.info("fstab says %s at %s is %s", dtype, file, ftype)
+            if fmt.test_mount():     # pylint: disable=no-member
+                device.format = fmt
+            else:
+                device.teardown()
+                raise ValueError(_(
```

Comment:
If this is a pre-existing entry, we might not want to raise an error here, it's probably not our job to tell users their fstab is wrong (unless they ask us explicitly to write something we think is wrong), especially because it simply might be correct and we just don't understand/support the filesystem.

Reply:
Changed Exception to log.info, so the unusual state is logged. If that happens the method will return None instead of a device.